src/theme/palette.ts

```diff
@@ -64,3 +65,76 @@ export type Palette = {
 } & Record<typeof mainTypes[number], generatedColorShades>;
 
 export type formFieldStyles = 'filled' | 'outlined' | 'elevated';
+
+/**
+ * this function picks either white or black color based on the background that is passed
+ * swatches are calculated based on accessibility by the design team and splited to those two colors
+ **/
+export const pickTextColorFromSwatches = (
+  color: typeof flatColors[number],
+  shade: typeof colorShades[number]
+) => {
+  const colorsForWhiteText: Partial<Record<
+    typeof colorShades[number],
+    typeof flatColors[number][]
+  >> = {
+    700: [
```

Comment:
Since these arrays are common you can export them in two const variables and concatenate them where it's needed.